### DISCUSSION TITLE
Lock application with code

### DISCUSSION THREAD

**Discussion by dotcs** [TechbeeAT/jtxBoard#204] [Ideas]
As the app might contain sensitive data, it would be great if it could be optionally locked with a pin, passphrase or pattern.

**Comment by patrickunterwegs** [comment]
Hey @dotcs , 
a similar request was made some time ago (https://gitlab.com/techbeeat1/jtx/-/issues/170).
A possible solution would be to use the device biometrics to show/hide entries, that are marked as "Confidential". Would this cover your request?

**Reply by dotcs** [reply]
Thanks for the quick reply. I was thinking more towards a global lock for the whole application as in my case most, if not all, journal entries and notes are only meant for my eyes. 

If you want to solve this on an individual item level, I would also like to see a default setting for the confidentiality level so that I don't have to enable the setting individually for each and every item I write.

Personally I would find a global lock on an application level a better solution.

**Reply by patrickunterwegs** [reply]
Alright, thanks, I'll follow this once up in #205 
I would add an option to choose wheter 
- None
- Confidential
- Confidential & Private
- All

entries are protected/only visible after confirmation through a fingerprint or face-id.

A dedicated block to open an app would be a thing for Android itself. Actually some Android-versions (e.g. from OnePlus) have an option to hide apps that can only be accessed after unlocking them through a fingerprint.
